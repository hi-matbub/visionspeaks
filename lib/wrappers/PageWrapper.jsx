import React from 'react';
import PropTypes from 'prop-types';
import Navigation from '../prebuilts/Navigation';
import Footer from '../prebuilts/Footer';

const PageWrapper = ({
  children,
  data,
  cart,
  handleUpdatedCartInState,
  currentBill,
}) => (
  <div className="content">
    <Navigation
      data={data}
      cart={cart}
      currentBill={currentBill}
      handleUpdatedCartInState={(updatedCart) =>
        handleUpdatedCartInState(updatedCart)
      }
    />
    <main>{children}</main>
    <Footer data={data} />
  </div>
);

PageWrapper.propTypes = {
  children: PropTypes.node.isRequired,
  cart: PropTypes.shape({}),
  data: PropTypes.shape({
    store: PropTypes.shape({}).isRequired,
  }).isRequired,
  handleUpdatedCartInState: PropTypes.func.isRequired,
  currentBill: PropTypes.number,
};

PageWrapper.defaultProps = {
  cart: {},
  currentBill: 0,
};

export default PageWrapper;
