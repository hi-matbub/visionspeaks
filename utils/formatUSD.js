/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  // for inconsistent types
  const string = JSON.stringify(parseFloat(number));
  const { length } = string || number;
  const str = `${string.substring(0, length - 2)}.${string.substring(
    length - 2
  )}`;
  return parseFloat(str).toFixed(2);
};

export { formatUSD };
