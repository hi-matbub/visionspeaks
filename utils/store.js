exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.svg',
  hero: '/hero.svg',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
