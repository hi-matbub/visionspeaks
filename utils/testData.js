// used for testing purposes

exports.testData = {
  store: {
    name: 'Dog-Gone Goods',
    logo: '/logo.svg',
    hero: '/hero.svg',
    url: '',
    description: 'A dog treat shop',
    about: 'Everyone loves a happy dog!',
    contact: 'youremail@here.com',
    social: { facebook: '', twitter: '', instagram: '' },
  },
  products: [
    {
      id: 'prod_IRbpyL6fgse3n',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151601,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Large Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153202,
    },
    {
      id: 'prod_IRbouvicOUom16',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151551,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153221,
    },
  ],
};
