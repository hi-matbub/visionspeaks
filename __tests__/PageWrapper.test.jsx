import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import PageWrapper from '../lib/wrappers/PageWrapper';
import { testData } from '../utils/testData';

const handleEmptyCartNotice = () => {}
const handleUpdatedCartInState = () => {}

it('PageWrapper renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <PageWrapper 
      data={testData}
      handleEmptyCartNotice={handleEmptyCartNotice}
      handleUpdatedCartInState={handleUpdatedCartInState}
    >
      <p>Hello World</p>
    </PageWrapper>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('PageWrapper renders unchanged', () => {
  const tree = renderer.create(
    <PageWrapper 
      data={testData}
      handleEmptyCartNotice={handleEmptyCartNotice}
      handleUpdatedCartInState={handleUpdatedCartInState}
    >
      <p>Hello World</p>
    </PageWrapper>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
