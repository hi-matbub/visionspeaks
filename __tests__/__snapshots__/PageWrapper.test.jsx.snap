// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper renders unchanged 1`] = `
<div
  className="content"
>
  <div
    className="d-flex justify-content-between w-100 fixed-top"
    style={
      Object {
        "background": "#FFFFFF !important",
      }
    }
  >
    <nav
      className="justify-content-between fixed-top navbar navbar-expand-lg navbar-light fixed-top"
      style={
        Object {
          "backgroundColor": "#FFFFFF !important",
        }
      }
    >
      <img
        alt="React Bootstrap logo"
        className="d-inline-block align-top"
        height="75"
        src="/logo.svg"
      />
      <a
        className="navbar-brand"
        href="/"
      >
        Dog-Gone Goods
      </a>
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "border": "0",
            }
          }
          type="button"
        >
          <div
            className="bg-warning scale-in-center"
            style={
              Object {
                "borderRadius": "50%",
                "height": "15px",
                "position": "absolute",
                "right": "14px",
                "top": "28px",
                "width": "15px",
              }
            }
          />
          <img
            alt="Checkout"
            src="/cart.png"
            style={
              Object {
                "cursor": "pointer",
                "height": "20px",
              }
            }
          />
        </button>
      </div>
    </nav>
  </div>
  <main>
    <p>
      Hello World
    </p>
  </main>
  <footer>
    <div
      className="my-5 container"
    >
      <div
        className="align-items-center justify-content-center row"
      >
        <div
          className="col-lg-8 col-md-9 col-sm-12"
        >
          <h2>
            More Info
          </h2>
          <div
            className="d-flex flex-wrap"
          >
            <div
              className="d-flex flex-column"
            >
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="About"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                About
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="FAQ"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                FAQ
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Contact"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Contact
              </button>
            </div>
            <div
              className="d-flex flex-column"
            >
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Terms and Conditions"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Terms and Conditions
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Privacy Policy"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Privacy Policy
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Shipping and Returns"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Shipping and Returns
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="d-flex justify-content-center col"
        >
          <p
            className="m-5 text-center"
          >
            Copyright 
            <br />
             
            2020
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
